src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -682,6 +682,34 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
                 }
             }
 
+            // If we call an unconstructed generic local function with a dynamic argument, we need to
+            // dynamically dispatch the call (as the function must be constructed at runtime).
+            // Check for the situation where a dynamic argument influences a type parameter,
+            // and disallow it.
+            if (boundMethodGroup.TypeArgumentsOpt.IsDefaultOrEmpty && localFunction.IsGenericMethod)
+            {
+                var originalLocalFunction = localFunction.OriginalDefinition;
+                for (int i = 0; i < args.Length; ++i)
+                {
+                    var argument = args[i];
+                    var parameter = originalLocalFunction.Parameters[methodResult.ParameterFromArgument(i)];
+                    if (argument.HasDynamicType() &&
+                        parameter.Type.ContainsTypeParameter(parameterContainer: originalLocalFunction))
```

Comment:
Is it necessary that the corresponding parameter type contains a type parameter. It looks like the following fails at runtime.
```
        Local((dynamic)"", "");
        void Local<T>(object x, T y) { }
```